PL2/PL2-shared/src/main/java/nl/tudelft/pl2016gr2/model/Node.java

```diff
@@ -55,12 +56,38 @@ public int hashCode() {
     return hash;
   }
 
-  public ArrayList<String> getGenomes() {
-    return genomes;
+  public Collection<String> getGenomes() {
+    return genomes.values();
   }
 
-  public void setGenomes(ArrayList<String> gs) {
-    this.genomes = gs;
+  /**
+   * Get the amount of genomes which travel from this node over an edge to the given node.
+   *
+   * @param to the node to which the edge goes.
+   * @return the amount of genomes which travel over the given edge.
+   */
+  @Override
+  public int getGenomesOverEdge(AbstractNode to) {
+    assert getOutlinks().contains(to.getId());
```

Comment:
Assert's are considered bad practice. Instead, you could either return 0 when there is no outlink to the other node, or throw an exception. 
Also, to be 100% secure you should also check if the node where it's going to contains an inlink with this node.